common/sgx/endorsements.c

```diff
@@ -302,14 +302,21 @@ oe_result_t oe_parse_sgx_endorsements(
         uint8_t* item_ptr = data_ptr_start + offsets[i];
         uint32_t item_size;
 
-        if (offsets[i] >= endorsements->buffer_size)
+        if (offsets[i] >= endorsements->buffer_size || offsets[i] >= data_size)
```

Comment:
`data_size = endorsements->buffer_size - offsets_size;
where offsets_size = OE_SGX_ENDORSEMENT_COUNT * (uint32_t)sizeof(uint32_t);`

Therefore, `data_size` is always smaller than `endorsements->buffer_size`. You only need one condition check here.